src/twisted/newsfragments/9343.feature

```diff
@@ -0,0 +1 @@
+twisted.python.compat.toBytes is a convenience function for converting unicode strings to bytes.
```

Comment:
Putting it in `twisted.python.compat` suggests this is only to do with bridging the Python 2/3 divide.  And so it will eventually go away because some behavior it offers is native to or natural in Python 3 ... but I don't think this is actually the case.  What Python 3-only construct idiomatically replaces `toBytes`?

Reply:
This does not need to live in `twisted.python.compat`, since this is a generally useful utility function for converting unicode to bytes, and should not be tied to `twisted.python.compat` which is a "temporary" bridging for Python 2/3.

Where is a better place for it `twisted.python.util`?

Reply:
`toBytes()` can be used all over the place.  There are places in `twisted.web` where it could be used, and would help eliminate Python 3-porting problems I have seen in 3rd party code which use twisted.web.  For example, here is one failure I found in txkube due to use of `networkString`:

https://travis-ci.org/LeastAuthority/txkube/jobs/304997908#L945

Reply:
`networkString` raising that exception may indicate there's a _bug_ in txkube though.  Silently converting between unicode and bytes when the programmer didn't _intend_ to it is one of the worst features of Python 2.  `networkString` fails in this case on purpose.  If `networkString` were replaced with an API like `toBytes` then it would be harder to find this kind of programming error (because the exception would not necessarily be raised).  So I'd recommend being very careful where this conversion function is used: it could easily take us back to the bad-old-times of carelessly mixed bytes/unicode.